Lib/ast.py

```diff
@@ -788,7 +788,12 @@ def get_raw_docstring(self, node):
             return node
 
     def get_type_comment(self, node):
-        comment = self._type_ignores.get(node.lineno) or node.type_comment
+        comment = None
+        if node.type_comment is not None:
+            comment = node.type_comment
+        elif (lineno := getattr(node, 'lineno', None)) is not None:
+            comment = self._type_ignores.get(lineno)
+
```

Comment:
Let's back up a little.

Assuming `node.lineno` exists, the old code behaved like this:
- if there is a type ignore for the current line, use it
- otherwise, use whatever is in `node.type_comment`

With your new code, `node.type_comment` is always used if it exists, reversing the priority.

Reply:
Yes, and this was intentional. If there is a type_comment attached, it should take the priority since `type_ignores` is a property of the module but on the other hand `type_comment` directly belongs to the node itself, so a change in it probably reflect / overwrite the specified `type_ignores`. Not sure if this deserves another PR with only this change (with still forcing for `lineno`), but I feel like `closer to the node == better reflectiveness` should hold up in this case.